### ISSUE TITLE
PHP 8 Support

### ISSUE THREAD

**Issue by DimaVIII** [webinarium/linode-api#3]
Hello @webinarium,

I'm trying to port is awesome library to php 8 but SplFixedArray class gives me a hard time.

This functions are deprecated 
```
SplFixedArray
    ::rewind()
    ::current()
    ::key()
    ::next()
    ::valid()
```
The new way with implements of Iterator, IteratorAggregate, RecursiveIterator are hard as I'm not familiar with the codebase. 
Some input 
- https://github.com/php/php-src/issues/8156 
- https://markbakeruk.net/2020/11/17/under-the-radar-a-backwards-compatible-break-for-splfixedarray-in-php-8/
I opened a fork and bumped versions in composer, everything seams to work in php 8 
https://github.com/DimaVIII/linode-api/tree/php8

It would be nice if you could take a look at this.

**Comment by webinarium** [comment]
Hello @DimaVIII 

I have released the new version 2.0.0, which is upgraded up to PHP 8.1.

**Comment by DimaVIII** [comment]
@webinarium Thank you! Awesome work! 
I will implement the library in my project and report if I find any bugs.
